zkevm-circuits/src/evm_circuit/op_execution/comparator.rs

```diff
@@ -0,0 +1,943 @@
+        ];
+
+        let success = {
+            // interpreter state transition constraints
+            let state_transition_constraints = vec![
+                state_next.global_counter.expr()
+                    - (state_curr.global_counter.expr() + 3.expr()),
+                state_next.stack_pointer.expr()
+                    - (state_curr.stack_pointer.expr() + 1.expr()),
+                state_next.program_counter.expr()
+                    - (state_curr.program_counter.expr() + 1.expr()),
+                state_next.gas_counter.expr()
+                    - (state_curr.gas_counter.expr() + GasCost::FASTEST.expr()),
+            ];
+
+            let LtSuccessAllocation {
+                selector,
+                swap,
+                a,
+                b,
+                result,
+                c,
+                carry,
+                sumc,
+                sumc_inv,
+            } = &self.success;
+
+            let no_swap = 1.expr() - swap.expr();
+            let swap_constraints = vec![
+                swap.expr() * no_swap.clone(),
+                swap.expr() * (opcode.expr() - OpcodeId::GT.expr()),
+                no_swap.clone() * (opcode.expr() - OpcodeId::LT.expr()),
+            ];
+
+            // sumc_expr == sumc_cell
+            // result = 1, sumc_expr != 0
+            let mut sumc_expr = 0.expr(); // The sum of c limbs
+            for byte in c.iter() {
+                sumc_expr = sumc_expr + byte.expr();
+            }
+            let sumc_is_zero_expr =
+                1.expr() - sumc_expr.clone() * sumc_inv.expr();
+            // same as is_zero gadget, currently we cannot use gadget inside OpGadget
+            let sumc_is_zero_constraints = vec![
+                sumc_expr.clone() * sumc_is_zero_expr.clone(),
+                sumc_inv.expr() * sumc_is_zero_expr.clone(),
+            ];
+            let sumc_constraints = vec![
+                sumc_expr - sumc.expr(),
+                result.cells[0].expr() * sumc_is_zero_expr.clone(),
+            ];
+
+            let mut lt_constraints = vec![];
+            // a[15..0] + c[15..0] = carry * 256^16 + b[15..0]
+            let mut exponent = 1.expr();
+            let mut lhs = 0.expr();
+            let mut rhs = 0.expr();
+            for idx in 0..16 {
+                lhs = lhs
+                    + (a.cells[idx].expr() + c[idx].expr()) * exponent.clone();
+                rhs = rhs + b.cells[idx].expr() * exponent.clone();
+                exponent = exponent * 256.expr();
+            }
+            rhs = rhs + carry.expr() * exponent;
+            lt_constraints.push(lhs - rhs);
+
+            // a[31..16] + c[31..16] + carry =
+            //     b[31..16] + (1 - result) * 256^16 * (1 - is_zero(sumc))
+            // - a < b, a + c = b, result = 1, is_zero(sumc) = 0
+            // - a = b, a + c = b, result = 0, is_zero(sumc) = 1
+            // - a > b, a + c = b + 2^256, result = 0, is_zero(sumc) = 0
```

Comment:
Nitpick: Using minus sign for bullet points could be confusing sometimes. 
```suggestion
            // a < b, a + c = b, result = 1, is_zero(sumc) = 0
            // a = b, a + c = b, result = 0, is_zero(sumc) = 1
            // a > b, a + c = b + 2^256, result = 0, is_zero(sumc) = 0
```

Reply:
Updated.